src/cascadia/TerminalSettingsEditor/NewTabMenuViewModel.cpp

```diff
@@ -0,0 +1,765 @@
+                }
+                break;
+            }
+            case NewTabMenuEntryType::Action:
+            {
+                if (const auto& actionEntry = entry.as<Model::ActionEntry>())
+                {
+                    result.push_back(make<ActionEntryViewModel>(actionEntry, settings));
+                }
+                break;
+            }
+            case NewTabMenuEntryType::Separator:
+            {
+                if (const auto& separatorEntry = entry.as<Model::SeparatorEntry>())
+                {
+                    result.push_back(make<SeparatorEntryViewModel>(separatorEntry));
+                }
+                break;
+            }
+            case NewTabMenuEntryType::Folder:
+            {
+                if (const auto& folderEntry = entry.as<Model::FolderEntry>())
+                {
+                    // The ctor will convert the children of the folder to view models
+                    result.push_back(make<FolderEntryViewModel>(folderEntry, settings));
+                }
+                break;
+            }
+            case NewTabMenuEntryType::MatchProfiles:
+            {
+                if (const auto& matchProfilesEntry = entry.as<Model::MatchProfilesEntry>())
+                {
+                    result.push_back(make<MatchProfilesEntryViewModel>(matchProfilesEntry));
+                }
+                break;
+            }
+            case NewTabMenuEntryType::RemainingProfiles:
+            {
+                if (const auto& remainingProfilesEntry = entry.as<Model::RemainingProfilesEntry>())
+                {
+                    result.push_back(make<RemainingProfilesEntryViewModel>(remainingProfilesEntry));
+                }
+                break;
+            }
+            case NewTabMenuEntryType::Invalid:
+            default:
+                break;
+            }
+        }
+        return single_threaded_observable_vector<Editor::NewTabMenuEntryViewModel>(std::move(result));
+    }
+
+    bool NewTabMenuViewModel::IsRemainingProfilesEntryMissing() const
+    {
+        return _IsRemainingProfilesEntryMissing(_rootEntries);
+    }
+
+    bool NewTabMenuViewModel::_IsRemainingProfilesEntryMissing(const IVector<Editor::NewTabMenuEntryViewModel>& entries)
+    {
+        for (const auto& entry : entries)
+        {
+            switch (entry.Type())
+            {
+            case NewTabMenuEntryType::RemainingProfiles:
+            {
+                return false;
+            }
+            case NewTabMenuEntryType::Folder:
+            {
+                if (!_IsRemainingProfilesEntryMissing(entry.as<Editor::FolderEntryViewModel>().Entries()))
+                {
+                    return false;
+                }
+                break;
+            }
+            default:
+                break;
+            }
+        }
+        return true;
+    }
+
+    bool NewTabMenuViewModel::IsFolderView() const noexcept
+    {
+        return CurrentView() != _rootEntries;

```

Comment:
nit: probably more readable if this is `return _CurrentFolder != nullptr`? Took me a bit to understand what this was used for haha